### PR TITLE
migrate odata client tests to xUnit

### PR DESCRIPTION
**Pull request by ElizabethOkerio**
### Issues

*This pull request fixes issue #xxx.*

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

**Comment by xuzhg** [comment]
>     <PackageReference Include="FluentAssertions" Version="4.1.0" />

shall we remove "FluentAssertions" since we move to xunit.net

---
Refers to: test/EndToEndTests/Tests/Client/Build.Desktop/Microsoft.Test.OData.Tests.Client.csproj:37 in 14d8b45. [](commit_id = 14d8b452f50ab7d138b20117627a6051a3988dc1, deletion_comment = False)

**Comment by xuzhg** [comment]
>         }

please move to top

---
Refers to: test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/Materialization/MaterializationPolicyUnitTests.cs:36 in 14d8b45. [](commit_id = 14d8b452f50ab7d138b20117627a6051a3988dc1, deletion_comment = False)

**Comment by xuzhg** [comment]
>     <PackageReference Include="FluentAssertions" Version="2.0.0.0" />

can we remove this dependency?

---
Refers to: test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Microsoft.OData.Client.TDDUnitTests.csproj:26 in 14d8b45. [](commit_id = 14d8b452f50ab7d138b20117627a6051a3988dc1, deletion_comment = False)

**Comment by xuzhg** [comment]
please rebase to latest master, make build pass and merge with squash

File: test/EndToEndTests/Tests/Client/AsynchronousTests/AsyncMethodTests.cs [hunk 1]
```diff
@@ -105,7 +105,7 @@ public async Task SaveChangesTest()
             customers.Add(c2);
 
             var dataServiceResponse = await context.SaveChangesAsync(SaveChangesOptions.BatchWithIndependentOperations | SaveChangesOptions.UseRelativeUri);
-            Assert.Equal((dataServiceResponse.First() as ChangeOperationResponse).StatusCode, 201);
+            Assert.Equal(201, (dataServiceResponse.First() as ChangeOperationResponse).StatusCode);
```
Comment (odero) [inline test/EndToEndTests/Tests/Client/AsynchronousTests/AsyncMethodTests.cs:108]: While you're here, you might want to use the `HttpStatusCode` Enum, for readability
https://docs.microsoft.com/en-us/dotnet/api/system.net.httpstatuscode

File: test/EndToEndTests/Tests/Client/BindingTests.cs [hunk 2]
```diff
@@ -13,27 +13,23 @@ namespace Microsoft.Test.OData.Tests.Client
 
     using Microsoft.Test.OData.Services.TestServices;
     using Microsoft.Test.OData.Services.TestServices.AstoriaDefaultServiceReference;
-#if WIN8 || WINDOWSPHONE
-    using Microsoft.VisualStudio.TestPlatform.UnitTestFramework;
-#else
-    using Microsoft.VisualStudio.TestTools.UnitTesting;
-
-#endif
 #if !PORTABLELIB && !SILVERLIGHT
```
Comment (odero) [inline test/EndToEndTests/Tests/Client/BindingTests.cs:16]: Possible to remove these portablelib silverlight references while you are doing this?

File: test/EndToEndTests/Tests/Client/Build.Desktop/AnnotationTests/InstanceAnnotationTests.cs [hunk 3]
```diff
@@ -113,31 +112,31 @@ public void ComplexTypeInstanceAnnotation()
                                 {
                                     // Verify Annotation on Complex Type
                                     ODataInstanceAnnotation annotationOnHomeAddress = entry.InstanceAnnotations.Last();
-                                    Assert.AreEqual(string.Format("{0}.AddressType", TestModelNameSpace), annotationOnHomeAddress.Name);
-                                    Assert.AreEqual("Home", (annotationOnHomeAddress.Value as ODataPrimitiveValue).Value);
+                                    Assert.Equal(string.Format("{0}.AddressType", TestModelNameSpace), annotationOnHomeAddress.Name);
+                                    Assert.Equal("Home", (annotationOnHomeAddress.Value as ODataPrimitiveValue).Value);
 
                                     // TODO : Fix #625
```
Comment (odero) [inline test/EndToEndTests/Tests/Client/Build.Desktop/AnnotationTests/InstanceAnnotationTests.cs:118]: Could you check on this issue and probably close if it's been resolved?

File: test/EndToEndTests/Tests/Client/Build.Desktop/BatchRequestTests/BatchRequestWithRelativeUriTests.cs [hunk 4]
```diff
@@ -136,72 +135,72 @@ private void BatchRequestWithPayloadUriWritingOption(BatchPayloadUriOption optio
...
+                                                Assert.NotNull(entry);
                                                 statements.Add(entry);
                                                 break;
                                         }
                                     }
-                                    Assert.AreEqual(ODataReaderState.Completed, entryReader.State);
-                                    Assert.AreEqual(1, statements.Count);
-                                    Assert.AreEqual(103901001, statements[0].Properties.Single(p => p.Name == "StatementID").Value);
+                                    Assert.Equal(ODataReaderState.Completed, entryReader.State);
+                                    Assert.Equal(1, statements.Count);
```
Comment (odero) [inline test/EndToEndTests/Tests/Client/Build.Desktop/BatchRequestTests/BatchRequestWithRelativeUriTests.cs:195]: Assert.Single?
```suggestion
                                    Assert.Single(statements);

```

File: test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientUpdateTests.cs [hunk 5]
```diff
@@ -42,17 +42,17 @@ public void BatchTest()
...
             var context = this.CreateWrappedContext().Context;
             var customerCount = context.Customer.Count();
 
             var customers = new DataServiceCollection<Customer>(context, context.Customer.GetAllPages(), TrackingMode.AutoChangeTracking, null, null, null);
-            Assert.AreEqual(customerCount, customers.Count());
+            Assert.Equal(customerCount, customers.Count());
             context.Configurations.RequestPipeline.OnEntryEnding((args) =>
             {
-                Assert.AreEqual(1, args.Entry.Properties.Count());
+                Assert.Equal(1, args.Entry.Properties.Count());
```
Comment (odero) [inline test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientUpdateTests.cs:55]:
```suggestion
                Assert.Single(args.Entry.Properties);

```

File: test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientUpdateTests.cs [hunk 6]
```diff
@@ -81,7 +81,7 @@ public void TrackEntitiesInAllPages()
             //context.LoadPropertyAllPages(customers[0], "Orders");
             //for (int i = 0; i < customers[0].Orders.Count(); i++)
             //{
-            //    Assert.AreEqual(customers[0].Orders[i].Concurrency.Token, "Order_ConCurrency_" + i.ToString());
+            //    Assert.Equal(customers[0].Orders[i].Concurrency.Token, "Order_ConCurrency_" + i.ToString());
```
Comment (odero) [inline test/EndToEndTests/Tests/Client/Build.Desktop/ClientTests/ClientUpdateTests.cs:84]: Clean out these commented lines

File: test/EndToEndTests/Tests/Client/Build.Desktop/ClientWithoutTypeResolverTests/JsonWithoutTypeResolverTests.cs [hunk 7]
```diff
@@ -86,26 +86,26 @@ public void NestedComplexPropertyQuery()
...
         }
 
-        [TestMethod]
+        [Fact]
         public void ServiceOperationFeedQuery()
         {
             var contextWrapper = this.CreateContext();
             var queryResult = contextWrapper.Execute<Customer>(new Uri(this.ServiceUri.OriginalString + "/GetSpecificCustomer?Name='enumeratetrademarkexecutionbrfalsenesteddupoverflowspacebarseekietfbeforeobservedstart'"), "GET", true).ToArray();
-            Assert.AreEqual(1, queryResult.Count(), "Expected a single Customer return");
+            Assert.Equal(1, queryResult.Count());
```
Comment (odero) [inline test/EndToEndTests/Tests/Client/Build.Desktop/ClientWithoutTypeResolverTests/JsonWithoutTypeResolverTests.cs:108]:
```suggestion
            Assert.Single(queryResult);

```

Comment (ElizabethOkerio) [reply test/EndToEndTests/Tests/Client/Build.Desktop/ClientWithoutTypeResolverTests/JsonWithoutTypeResolverTests.cs:108]: xunit recommends the use of Single for a collection with a count of 1.

Comment (odero) [reply test/EndToEndTests/Tests/Client/Build.Desktop/ClientWithoutTypeResolverTests/JsonWithoutTypeResolverTests.cs:108]: Isn't that what this is?

**Pull request merged**